test/pylib: scylla_cluster: set endpoint_snitch in scylla conf

bhalevy opened the pull request:
When `property_file` is provided, we generate a
`cassandra-rackdc.properties` file, but to actually use it, `endpoint_snitch` must be set to `GossipingPropertyFileSnitch`.

* No backport required at this point

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11772/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11772/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_custom/test_snitch_change
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11772/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11772/)
#### Build Details:
- Duration: 7 hr 6 min
- Builder: i-09046de47c030eeb5 (m5ad.8xlarge)

xemul commented on test/pylib/manager_client.py:
```diff
@@ -374,20 +379,21 @@ async def servers_add(self, servers_num: int = 1,
                                                       timeout=ScyllaServer.TOPOLOGY_TIMEOUT * servers_num)
         except Exception as exc:
-            raise Exception("Failed to add servers") from exc
+            raise Exception(f"Failed to add servers: {exc}") from exc
```
Doesn't raise-from includes the original `exc` into what's thrown in verbose enough manner?

bhalevy replied:
yes, I forgot about it

bhalevy replied:
It somehow wasn't very helpful during debugging.
Maybe it was just a pilot error.
Anyhow, I'll remove this hunk and rebase

xemul commented on test/pylib/scylla_cluster.py:
```diff
@@ -298,6 +304,7 @@ def __init__(self, mode: str, exe: str, vardir: str,
         self.resourcesdir = pathlib.Path.cwd() / "test/pylib/resources"
         self.resources_certificate_file = self.resourcesdir / "scylla.crt"
         self.resources_keyfile_file = self.resourcesdir / "scylla.key"
+        self.endpoint_snitch = endpoint_snitch or "GossipingPropertyFileSnitch" if property_file else None
```
So this patch not only sets endpoint-snitch if property-file is provided, but also allows specifying some "custom" endpoint-snitch config?

bhalevy replied:
yes, and `test_snitch_change` actually uses that since it relied on the snitch to remain as default despite providing a property_file

xemul replied:
OK. The new argument for manager::server_add() looks quite a big hammer against inability to setup custom snitch nail :( Can't snitch name be passed via existing config or cmdline argument?

bhalevy replied:
Hmm, good point.  I think this is feasible.
I'll take a stab at it

bhalevy replied:
done in next version

bhalevy commented:
[82e5138](https://github.com/scylladb/scylladb/commit/82e5138cc9baaac3831ff90545ec3337f1d47975):
- rebased
- dropped hunk changing formatting of Exception

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11883/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11883/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_custom/test_snitch_change
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11883/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11883/)
#### Failed Tests (1/27742):
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11883/testReport/junit/boost.cql_query_test.test_drop_table_with_si_and_mv.1.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 7 hr 18 min
- Builder: i-0523e990f14cf0075 (m5ad.8xlarge)

bhalevy commented:
In [14d540b](https://github.com/scylladb/scylladb/commit/14d540b77b796531fa91f43a0e33e70734d7484c):
- Allow overriding the default endpoint_snitch using the config dict
  - Use in topology_custom/test_snitch_change

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11905/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11905/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: topology_custom/test_snitch_change
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11905/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11905/)
#### Build Details:
- Duration: 7 hr 20 min
- Builder: i-0ffef5a4d532b0c30 (m5ad.8xlarge)

bhalevy commented:
clang-tidy is bitching about existing code
that we aren't going to change any time soon.
I'm not sure what value we have from running it in CI...

bhalevy commented:
@xemul please re-review / merge
